tests/testthat/test-redundant_ifelse_linter.R

```diff
@@ -0,0 +1,79 @@
+test_that("redundant_ifelse_linter skips allowed usages", {
+  expect_lint("ifelse(x > 5, 0, 2)", NULL, redundant_ifelse_linter())
+  expect_lint("ifelse(x > 5, TRUE, NA)", NULL, redundant_ifelse_linter())
+  expect_lint("ifelse(x > 5, FALSE, NA)", NULL, redundant_ifelse_linter())
+  expect_lint("ifelse(x > 5, TRUE, TRUE)", NULL, redundant_ifelse_linter())
+
+  expect_lint("ifelse(x > 5, 0L, 2L)", NULL, redundant_ifelse_linter())
+  expect_lint("ifelse(x > 5, 0L, 10L)", NULL, redundant_ifelse_linter())
+})
+
+test_that("redundant_ifelse_linter blocks simple disallowed usages", {
+  expect_lint(
+    "ifelse(x > 5, TRUE, FALSE)",
+    rex::rex("Just use the logical condition (or its negation) directly"),
+    redundant_ifelse_linter()
+  )
+  expect_lint(
+    "ifelse(x > 5, FALSE, TRUE)",
+    rex::rex("Just use the logical condition (or its negation) directly"),
+    redundant_ifelse_linter()
+  )
+
+  expect_lint(
+    "ifelse(x > 5, 1L, 0L)",
+    rex::rex("Prefer as.integer(x) to ifelse(x, 1L, 0L)"),
+    redundant_ifelse_linter()
+  )
+  expect_lint(
+    "ifelse(x > 5, 0L, 1L)",
+    rex::rex("Prefer as.integer(x) to ifelse(x, 0L, 1L)"),
+    redundant_ifelse_linter()
+  )
+
+  expect_lint(
+    "ifelse(x > 5, 1, 0)",
+    rex::rex("Prefer as.numeric(x) to ifelse(x, 1, 0)"),
```

Comment:
Nit: IINM 1 and 0 are implicitly integers.

Reply:
```
typeof(ifelse(runif(10) > .5, 1, 0))
# [1] "double"
```

?